src/main/kotlin/org/rust/lang/refactoring/RsNamesValidator.kt

```diff
@@ -18,10 +19,10 @@ class RsNamesValidator : NamesValidator {
         isIdentifier(name, project, true)
 
     fun isIdentifier(name: String, @Suppress("UNUSED_PARAMETER") project: Project?, withPrimitives: Boolean): Boolean =
-        if (withPrimitives) {
-            getLexerType(name) == IDENTIFIER && name !in PrimitiveTypes
-        } else {
-            getLexerType(name) == IDENTIFIER
+        when (getLexerType(name)) {
+            IDENTIFIER -> !withPrimitives || name !in PrimitiveTypes
+            QUOTE_IDENTIFIER -> name !in PredefinedLifetimes
```

Comment:
>And its declaration could be removed after renaming to 'static with help of RenamePsiElementProcessor

That's a bit to magical I think. Let's stick with just renaming to `'static` for now, and then add conflicts dialog like the one in Kotlin?